accepted/0000-sle-micro-reboot-action.md

```diff
@@ -0,0 +1,58 @@
+
+The user must be able to change reboot method at any time and SUSE Manager will keep it. In any case, we should alert in the documentation that using any method different from `systemd` may cause undesired behavior.
+
+### Disable `transactional-update.timer` and `rebootmgr` services.
+
+If the system was in its default configuration, and we changed the reboot method in bootstrap, we need to disable services that deals with reboot but are not controlled by Suma.
+
+By default, `transactional-update.timer` is configured to check for updates every day, and if updates are available and could be applied successfully it will reboot the system afterwards automatically. The configuration is on `/etc/systemd/system/timers.target.wants/transactional-update.timer`. To avoid undesired reboots in managed systems, this auto reboot service should be disabled.
+
+`rebootmgr` service allows the user to define maintenance windows where the pending reboots should be performed. This also can cause undesired behavior when the system is managed by Suma, so, if we change the reboot method to put Suma in control, we should also disable `rebootmgr` service.
+
+Both these services can be re-enabled by users if they change the reboot method to others that needed them to be enabled. SUSE Manager will not change the configuration of this service after onboard.
+
+### Use `transactional-update` module to perform the reboot
+
+In the end, whatever is configured for reboot method, Suma should perform reboot actions calling the [reboot](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.transactional_update.html#salt.modules.transactional_update.reboot) function of `transactional_update` salt module. If `systemd` is configured, the system will reboot immediately and Suma will be aware of the reboot. If other method is configured, Suma will only ask for a reboot and there is no guarantee of when this reboot will be performed. If the reboot doesn't take longer than 6 hours the action will be updated as successfully completed in Suma. But, if there is any peculiarity in the reboot method used (such as a specific maintenance window), and the reboot takes longer than 6 hours, the action will be updated as failed in Suma.
+
+This behavior should be documented to make customers aware with the advice either to stay with `systemd` and let Suma decide when to reboot or they should schedule reboot actions only at a time when their configured method allows to reboot.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Create a `suma` reboot method for `transactional-update`. If the user configure the system to use any other method, reboot wouldn't be allowed in Suma to avoid undesired behavior. This alternative would make it necessary to patch `transactional-update` and the only benefit seems to be having a guarantee that Suma only try to perform reboot if it is in control of it.
```

Comment:
Maybe it's useful to explain the main difference between "Uyuni triggers reboot via systemd method" and "Uyuni is a reboot method" in this part: If "Uyuni is a reboot method", all reboots must go through it. Including when a user connects to the client via `ssh` and runs `transactional-update reboot`. That requires Uyuni to keep track of requested reboots.